src/TriggerBinding/SqlTriggerBinding.cs

```diff
@@ -0,0 +1,196 @@
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Gets the type of the value the Trigger receives from the Executor.
+        /// </summary>
+        public Type TriggerValueType => typeof(IReadOnlyList<SqlChange<T>>);
+
+        /// <summary>
+        /// Returns an empty binding contract. The type that SqlTriggerAttribute is bound to is checked in 
+        /// <see cref="SqlTriggerAttributeBindingProvider.TryCreateAsync(TriggerBindingProviderContext)"/>
+        /// </summary>
+        public IReadOnlyDictionary<string, Type> BindingDataContract => _emptyBindingContract;
+
+        /// <summary>
+        /// Binds the list of <see cref="SqlChange{T}"/> represented by "value" with a <see cref="SimpleValueProvider"/>
+        /// which (as the name suggests) simply returns "value".
+        /// <param name="value">
+        /// The list of <see cref="SqlChange{T}"/> data.
+        /// </param>
+        /// <param name="context">
+        /// Unused
+        /// </param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if "value" is not of type IReadOnlyList<SqlChange<T>>.
+        /// </exception>
+        /// <returns>
+        /// The ITriggerData which stores the list of SQL table changes as well as the SimpleValueBinder
+        /// </returns>
+        public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+        {
+            if (!(value is IReadOnlyList<SqlChange<T>> changes))
+            {
+                throw new InvalidOperationException("The value passed to the SqlTrigger BindAsync must be of type IReadOnlyList<SqlChange<T>>");
+            }
+
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "SqlTrigger", changes }
+            };
+
+            return Task.FromResult<ITriggerData>(new TriggerData(new SimpleValueProvider(this._parameter.ParameterType, changes, this._tableName), bindingData));
+        }
+
+        /// <summary>
+        /// Creates a listener that will monitor for changes to the user's table
+        /// </summary>
+        /// <param name="context">
+        /// Context for the listener, including the executor that executes the user's function when changes are detected in the user's table
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if context is null
+        /// </exception>
+        /// <returns>
+        /// The listener
+        /// </returns>
+        public async Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context), "Missing listener context");
+            }
+
+            string userFunctionId = SqlBindingUtilities.AsSingleQuoteEscapedString(await this.GetUserFunctionIdAsync());
```

Comment:
`SqlBindingUtilities.AsSingleQuoteEscapedString` is called twice, once inside `GetUserFunctionIdAsync()` and once in the caller. Both are unnecessary, as method `GetUserFunctionIdAsync` will return a string with lower-case hex characters like `"36a6cc45b7bd4c83"`.